src/Rule/SubsetHandler.php

```diff
@@ -38,13 +33,7 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
                 : $rule->getValues();
             $valuesString = '"' . implode('", "', $values) . '"';
 
-            $result->addError(
-                $rule->getSubsetMessage(),
-                [
-                    'attribute' => $context->getAttribute(),
-                    'values' => $valuesString,
```

Comment:
`values` can be useful for custom messages.

Reply:
https://github.com/yiisoft/validator/pull/352#discussion_r1006452279

Reply:
https://github.com/yiisoft/validator/pull/352#discussion_r1006471638